source/features/rgh-improve-new-issue-form.tsx

```diff
@@ -5,12 +5,42 @@ import * as pageDetect from 'github-url-detection';
 import features from '../feature-manager';
 import openOptions from '../helpers/open-options';
 import clearCacheHandler from '../helpers/clear-cache-handler';
-import {expectTokenScope} from '../github-helpers/api';
+import {expectToken, expectTokenScope} from '../github-helpers/api';
 import {isRefinedGitHubRepo} from '../github-helpers';
+import {assertNodeContent} from '../helpers/dom-utils';
 
-function init(): void {
+function addNotice(adjective: JSX.Element | string): void {
+	select('#issue_body_template_name')!.before(
+		<div className="flash flash-warn m-2">{}
+			Your Personal Access Token is {adjective}. Some Refined GitHub features will not work without it.
+			You can update it <button className="btn-link" type="button" onClick={openOptions as unknown as React.MouseEventHandler}>in the options</button>.
+		</div>,
+	);
+}
+
+async function checkToken(): Promise<void> {
+	try {
+		await expectToken();
+	} catch {
+		addNotice('missing');
+		return;
+	}
+
+	try {
+		await expectTokenScope('repo');
+	} catch {
+		addNotice('invalid or expired');
```

Comment:
The API just returns `Bad credentials` so we can't know further details, I think.